index-scheduler/src/batch.rs

```diff
@@ -714,6 +711,14 @@ impl IndexScheduler {
                 // 5.3 Change the permission to make the snapshot readonly
                 let mut permissions = file.metadata()?.permissions();
                 permissions.set_readonly(true);
+                #[cfg(unix)]
+                {
+                    use std::os::unix::fs::PermissionsExt;
+                    #[allow(clippy::non_octal_unix_permissions)]
+                    //                     rwxrwxrwx
+                    permissions.set_mode(0b100100100);
```

Comment:
What is the mode set by`set_readonly`? not sure to understand what they meant by readonly.

Reply:
From what I see, it removes the right to write + sets the right to read if you're root.

_I think_ it's sufficient for windows, but it's not for UNIX. I decided to keep it just in case, but tbh, I don't really know if we should.

Reply:
Thank you! It looks like you are right. We must set the read bits to true for the root part.